### PR TITLE
Add world support

### PR DESCRIPTION
**Pull request by akshualy**
I needed world support for my server and thought I'd pull request my changes back to you.
Since the command to teleport to specific worlds is very awkward in vanilla Minecraft, I added an `xcatch tp` command.
I tried to follow your code style as closely as possible.

### CODE REVIEW

File: src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java [hunk 1]
```diff
@@ -136,17 +144,21 @@ public void onInventoryClick(InventoryClickEvent event) {
...
-                                    put("{x}", coordinates[0]);
-                                    put("{y}", coordinates[1]);
-                                    put("{z}", coordinates[2]);
+                                    put("{world}", XCatch.INSTANCE.getServer().getWorld(actionData.worldUID).getName());
+                                    put("{x}", String.valueOf(actionData.x));
+                                    put("{y}", String.valueOf(actionData.y));
+                                    put("{z}", String.valueOf(actionData.z));
                                 }};
                                 event.getWhoClicked().closeInventory();
+                                XCatch.INSTANCE.getLogger().info(Utils.replaceVariables(XCatch.config.getString("view-click-command"), variables));
```
Comment (DeDiamondPro) [inline src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java]: Is this log needed?

File: src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java [hunk 2]
```diff
@@ -136,17 +144,21 @@ public void onInventoryClick(InventoryClickEvent event) {
...
+                            );
+                            if (actionIndex == null) return;
+                            ActionData actionData = PersistentData.data.actions.get(uuid).get(actionIndex);
                             try {
                                 HashMap<String, String> variables = new HashMap<String, String>() {{
                                     put("{player}", name);
-                                    put("{x}", coordinates[0]);
-                                    put("{y}", coordinates[1]);
-                                    put("{z}", coordinates[2]);
+                                    put("{world}", XCatch.INSTANCE.getServer().getWorld(actionData.worldUID).getName());
```
Comment (DeDiamondPro) [inline src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java]: If the world is null (older versions) this won't work and backwards compatibility is very important for me
```suggestion
                                    put("{world}", actionData.worldUID == null ? event.getWhoClicked().getWorld() : XCatch.INSTANCE.getServer().getWorld(actionData.worldUID).getName());
```

**Review by DeDiamondPro — CHANGES_REQUESTED** [review]
Great addition although I do have a few small notes

File: src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java (hunk 1 above)
Comment (akshualy) [reply src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java]: Nope, forgot to remove it when debugging world names, good catch.

File: src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java (hunk 2 above)
Comment (akshualy) [reply src/main/java/io/github/dediamondpro/xcatch/gui/ViewGui.java]: Good point, I've also made sure with a quick self-test that the .json.gz file still loads if the worldUID is missing in previous version entries.

**Review by DeDiamondPro — APPROVED** [review]

**Pull request merged**